ros/docker/encoder_odom/Dockerfile

```
@@ -2,17 +2,28 @@ FROM ros:noetic-ros-base
 
 SHELL ["/bin/bash", "-c"]
 
+WORKDIR /app
+
 RUN apt update && apt install -y curl
 
+RUN apt-get -y update
+
+RUN apt-get install -y python3-tf2-ros
+
+RUN apt-get install -y python3-tf-conversions
```

Docker caches in between these RUN directives, meaning the `apt-get update` line could be cached and then the `apt-get install` line runs on a cached version of the apt lists. You need to run update in the same RUN directive:

```suggestion
RUN apt-get update && apt-get install -yq python3-tf2-ros python3-tf-conversions
```